src/gui/LayoutMapper.cpp

```diff
@@ -55,8 +35,13 @@ LayoutMapper::LayoutMapper(int numPages, Settings* settings)
 		pairsOffset = 0;
 	}
 
-	LayoutType type = compileLayout(isVertical, isRightToLeft, isBottomToTop);
-	layoutMapperInit(pages, numRows, numCols, fixRows, type, isPairedPages, pairsOffset);
+	//assemble bitflags for LayoutType
+	int type = isVertical ? LayoutBitFlags::Vertically : 0;
+	type |= isRightToLeft ? LayoutBitFlags::RightToLeft : 0;
+	type |= isBottomToTop ? LayoutBitFlags::BottomToTop : 0;
+
+	
+	layoutMapperInit(pages, numRows, numCols, fixRows, (LayoutType)type, isPairedPages, pairsOffset);
```

Comment:
You could merge `configureFromSettings()` and `layoutMapperInit()` as this is the only position where you call the method. But depending on what you want to add you can also leave those two methods separate. This is no real issue, just something you may think about.

Reply:
That could probably be cleaned up a bit more now that it's settled in. Probably best in another commit so I'll have to come back to it.